internal/relayenv/env_context.go

```diff
@@ -45,22 +50,9 @@ type EnvContext interface {
 	// GetDeprecatedCredentials returns all deprecated and not-yet-removed credentials for the environment.
 	GetDeprecatedCredentials() []credential.SDKCredential
 
-	// AddCredential adds a new credential for the environment.
-	//
-	// If the credential is an SDK key, then a new SDK client is started with that SDK key, and event forwarding
-	// to server-side endpoints is switched to use the new key.
-	AddCredential(credential.SDKCredential)
-
-	// RemoveCredential removes a credential from the environment. Any active stream connections using that
-	// credential are immediately dropped.
-	//
-	// If the credential is an SDK key, then the SDK client that we started with that SDK key is disposed of.
-	RemoveCredential(credential.SDKCredential)
-
-	// DeprecateCredential marks an existing credential as not being a preferred one, without removing it or
-	// dropping any connections. It will no longer be included in the return value of GetCredentials(). This is
-	// used in Relay Proxy Enterprise when an SDK key is being changed but the old key has not expired yet.
-	DeprecateCredential(credential.SDKCredential)
```

Comment:
Rather than speak about adding/removing/deprecating credentials, that level of detail is no longer exposed on the EnvContext interface.

Instead, components can ask the environment to rotate credentials. I've split them out into the actual mobile/env/SDK key types, because deprecation is actually only relevant for SDK keys. This makes it clear.